src/openforms/registrations/contrib/json_dump/tests/test_backend.py

```diff
@@ -0,0 +1,238 @@
+from base64 import b64decode
+from pathlib import Path
+
+from django.core.exceptions import SuspiciousOperation
+from django.test import TestCase
+
+from glom import glom
+from requests import RequestException
+from zgw_consumers.test.factories import ServiceFactory
+
+from openforms.submissions.public_references import set_submission_reference
+from openforms.submissions.tests.factories import (
+    SubmissionFactory,
+    SubmissionFileAttachmentFactory,
+)
+from openforms.utils.tests.vcr import OFVCRMixin
+
+from ..plugin import JSONDumpRegistration
+
+VCR_TEST_FILES = Path(__file__).parent / "files"
+
+
+class JSONDumpBackendTests(OFVCRMixin, TestCase):
+    VCR_TEST_FILES = VCR_TEST_FILES
+
+    def test_submission_happy_flow(self):
+        submission = SubmissionFactory.from_components(
+            [
+                {"key": "firstName", "type": "textfield"},
+                {"key": "lastName", "type": "textfield"},
+                {"key": "file", "type": "file"},
+            ],
+            completed=True,
+            submitted_data={
+                "firstName": "We Are",
+                "lastName": "Checking",
+                "file": [
+                    {
+                        "url": "some://url",
+                        "name": "my-foo.bin",
+                        "type": "application/foo",
+                        "originalName": "my-foo.bin",
+                    }
+                ],
+            },
+            bsn="123456789",
+        )
+
+        SubmissionFileAttachmentFactory.create(
+            form_key="file",
+            submission_step=submission.submissionstep_set.get(),
+            file_name="test_file.txt",
+            original_name="test_file.txt",
+            content_type="application/text",
+            content__data=b"This is example content.",
+            _component_configuration_path="components.2",
+            _component_data_path="file",
+        )
+
+        json_form_options = dict(
+            service=(ServiceFactory(api_root="http://localhost:80/")),
+            path="json_plugin",
+            variables=["firstName", "file", "auth_bsn"],
+        )
+        json_plugin = JSONDumpRegistration("json_registration_plugin")
+        set_submission_reference(submission)
```

Comment:
it's better DX to pass the `with_public_registration_reference=True` trait to the `SubmissionFactory` :)